--- joplin/packages/lib/database.ts	2022-12-08 18:39:09.232479187 +0000
+++ database.ts	2022-12-08 18:43:31.387167128 +0000
@@ -137,6 +137,10 @@
 
 				return result; // No exception was thrown
 			} catch (error) {
+				if (error && error.code == 'SQLITE_READONLY'){
+					alert("Joplin has been updated while running and must be restarted");
+					require("@electron/remote").app.exit(0);
+				}
 				if (error && (error.code === 'SQLITE_IOERR' || error.code === 'SQLITE_BUSY')) {
 					if (totalWaitTime >= 20000) throw this.sqliteErrorToJsError(error, sql, params);
 					// NOTE: don't put logger statements here because it might log to the database, which
