diff -Naur joplin/packages/app-desktop/commands/copyDevCommand.ts joplin-patched/packages/app-desktop/commands/copyDevCommand.ts
--- joplin/packages/app-desktop/commands/copyDevCommand.ts	2025-01-11 22:39:20.645638271 +0000
+++ joplin-patched/packages/app-desktop/commands/copyDevCommand.ts	2025-01-12 00:19:04.535121885 +0000
@@ -1,7 +1,6 @@
 import { CommandRuntime, CommandDeclaration } from '@joplin/lib/services/CommandService';
 import { _ } from '@joplin/lib/locale';
 import shim, { MessageBoxType } from '@joplin/lib/shim';
-const app = require('@electron/remote').app;
 const { clipboard } = require('electron');
 
 export const declaration: CommandDeclaration = {
@@ -12,7 +11,7 @@
 export const runtime = (): CommandRuntime => {
 	return {
 		execute: async () => {
-			const appPath = app.getPath('exe');
+			const appPath = process.env.SNAP_INSTANCE_NAME;
 			const cmd = `${appPath} --env dev`;
 			clipboard.writeText(cmd);
 			await shim.showMessageBox(`The dev mode command has been copied to clipboard:\n\n${cmd}`, { type: MessageBoxType.Info });
