--- joplin/packages/app-desktop/commands/copyDevCommand.ts	2021-10-01 21:47:35.072350714 +0100
+++ copyDevCommand.ts	2021-10-02 19:24:39.508602730 +0100
@@ -1,6 +1,5 @@
 import { CommandRuntime, CommandDeclaration } from '@joplin/lib/services/CommandService';
 import { _ } from '@joplin/lib/locale';
-const app = require('@electron/remote').app;
 const { clipboard } = require('electron');
 
 export const declaration: CommandDeclaration = {
@@ -11,7 +10,7 @@
 export const runtime = (): CommandRuntime => {
 	return {
 		execute: async () => {
-			const appPath = app.getPath('exe');
+			const appPath = process.env.SNAP_INSTANCE_NAME;
 			const cmd = `${appPath} --env dev`;
 			clipboard.writeText(cmd);
 			alert(`The dev mode command has been copied to clipboard:\n\n${cmd}`);
