--- joplin/packages/app-desktop/bridge.ts	2021-08-07 18:54:36.908134238 +0100
+++ bridge.ts	2021-08-07 20:00:29.652916579 +0100
@@ -166,7 +166,18 @@
 	}
 
 	openExternal(url: string) {
-		return require('electron').shell.openExternal(url);
+		return require("child_process").execFile("joplin-open", [url], (error: any, stdout: any, stderr: any) => {
+			if (error) {
+				console.log(`error: ${error.message}`);
+				return;
+			}
+			if (stderr) {
+				console.log(`stderr: ${stderr}`);
+				return;
+			}
+			console.log(stdout);
+			return;
+		});
 	}
 
 	async openItem(fullPath: string) {
