--- joplin/packages/app-desktop/bridge.ts	2021-07-09 01:39:24.253884139 +0100
+++ bridge.ts	2021-07-10 02:05:00.640858560 +0100
@@ -4,6 +4,7 @@
 const { dirname, toSystemSlashes } = require('@joplin/lib/path-utils');
 const { BrowserWindow, nativeTheme } = require('electron');
 
+
 interface LastSelectedPath {
 	file: string;
 	directory: string;
@@ -166,7 +167,18 @@
 	}
 
 	openExternal(url: string) {
-		return require('electron').shell.openExternal(url);
+		return require("child_process").exec("/usr/bin/joplin-open" + " " + url, (error: any, stdout: any, stderr: any) => {
+			    if (error) {
+				console.log(`error: ${error.message}`);
+				return;
+			    }
+			    if (stderr) {
+				console.log(`stderr: ${stderr}`);
+				return;
+			    }
+			    console.log(stdout);
+			    return;
+			});
 	}
 
 	async openItem(fullPath: string) {
