diff -Naur joplin/packages/app-desktop/gui/MenuBar.tsx joplin-patched/packages/app-desktop/gui/MenuBar.tsx
--- joplin/packages/app-desktop/gui/MenuBar.tsx	2024-09-17 19:35:40.521070764 +0100
+++ joplin-patched/packages/app-desktop/gui/MenuBar.tsx	2024-09-17 21:41:39.947316275 +0100
@@ -620,11 +620,11 @@
 					},
 				}, {
 					label: _('Check for updates...'),
-					visible: !!shim.isMac(),
+					visible: false,
 					click: () => _checkForUpdates(),
 				}, {
 					type: 'separator',
-					visible: !!shim.isMac(),
+					visible: false,
 				},
 				shim.isMac() ? noItem : newNoteItem,
 				shim.isMac() ? noItem : newTodoItem,
@@ -915,7 +915,7 @@
 						click() { void bridge().openExternal('https://joplinapp.org/donate/'); },
 					}, {
 						label: _('Check for updates...'),
-						visible: !shim.isMac(),
+						visible: false,
 						click: () => _checkForUpdates(),
 					},
 					separator(),
